gapis/api/vulkan/footprint_builder.go

```diff
@@ -0,0 +1,3283 @@
+		src := []dependencygraph.DefUseAtom{vb.getImageData(ctx, bh, cmd.SrcImage)}
+		overwritten := false
+		count := uint64(cmd.RegionCount)
+		regions := cmd.PRegions.Slice(0, count, l)
+		// TODO: check dst image coverage correctly
+		for i := uint64(0); i < count; i++ {
+			region := regions.Index(i, l).Read(ctx, cmd, s, nil)
+			overwritten = overwritten || subresourceLayersFullyCoverImage(
+				GetState(s).Images.Get(cmd.DstImage),
+				region.DstSubresource, region.DstOffset, region.Extent)
+		}
+		if overwritten {
+			vb.recordReadsWritesModifies(
+				ctx, ft, bh, cmd.CommandBuffer, src, dst, emptyAtoms)
+		} else {
+			vb.recordReadsWritesModifies(
+				ctx, ft, bh, cmd.CommandBuffer, src, emptyAtoms, dst)
+		}
+	case *RecreateCmdCopyImage:
+		dst := []dependencygraph.DefUseAtom{vb.getImageData(ctx, bh, cmd.DstImage)}
+		src := []dependencygraph.DefUseAtom{vb.getImageData(ctx, bh, cmd.SrcImage)}
+		overwritten := false
+		count := uint64(cmd.RegionCount)
+		regions := cmd.PRegions.Slice(0, count, l)
+		for i := uint64(0); i < count; i++ {
+			region := regions.Index(i, l).Read(ctx, cmd, s, nil)
+			overwritten = overwritten || subresourceLayersFullyCoverImage(
+				GetState(s).Images.Get(cmd.DstImage),
+				region.DstSubresource, region.DstOffset, region.Extent)
+		}
+		if overwritten {
+			vb.recordReadsWritesModifies(
+				ctx, ft, bh, cmd.CommandBuffer, src, dst, emptyAtoms)
+		} else {
+			vb.recordReadsWritesModifies(
+				ctx, ft, bh, cmd.CommandBuffer, src, emptyAtoms, dst)
+		}
+
+	case *VkCmdCopyBuffer:
+		srcBufferData, _ := vb.getBufferData(ctx, bh, cmd.SrcBuffer).(memorySpan)
+		dstBufferData, _ := vb.getBufferData(ctx, bh, cmd.DstBuffer).(memorySpan)
+		src := []dependencygraph.DefUseAtom{}
+		dst := []dependencygraph.DefUseAtom{}
+		count := uint64(cmd.RegionCount)
+		regions := cmd.PRegions.Slice(0, count, l)
+		for i := uint64(0); i < count; i++ {
+			region := regions.Index(i, l).Read(ctx, cmd, s, nil)
+			srcOffset := srcBufferData.span.Start + uint64(region.SrcOffset)
+			dstOffset := dstBufferData.span.Start + uint64(region.DstOffset)
+			src = append(src, memorySpan{
+				span:   interval.U64Span{Start: srcOffset, End: srcOffset + uint64(region.Size)},
+				memory: srcBufferData.memory})
+			dst = append(dst, memorySpan{
+				span:   interval.U64Span{Start: dstOffset, End: dstOffset + uint64(region.Size)},
+				memory: dstBufferData.memory})
+		}
+		vb.recordReadsWritesModifies(
+			ctx, ft, bh, cmd.CommandBuffer, src, dst, emptyAtoms)
+	case *RecreateCmdCopyBuffer:
+		srcBufferData, _ := vb.getBufferData(ctx, bh, cmd.SrcBuffer).(memorySpan)
+		dstBufferData, _ := vb.getBufferData(ctx, bh, cmd.DstBuffer).(memorySpan)
+		src := []dependencygraph.DefUseAtom{}
+		dst := []dependencygraph.DefUseAtom{}
+		count := uint64(cmd.RegionCount)
+		regions := cmd.PRegions.Slice(0, count, l)
+		for i := uint64(0); i < count; i++ {
+			region := regions.Index(i, l).Read(ctx, cmd, s, nil)
+			srcOffset := srcBufferDa
```

Comment:
Nope, we cannot do that. This switch is a type switch, we cannot access the members of the concrete api.Cmd. Actually I was thinking of this before, as all the vkCmdxxx  and RecreateCmdxxx commands have exactly the same code. But unfortunately, we cannot do that.